src/map.cpp

```diff
@@ -532,62 +532,150 @@ bool Map::canThrowObjectTo(const Position& fromPos, const Position& toPos, bool
 	return isSightClear(fromPos, toPos, false);
 }
 
-bool Map::checkSightLine(const Position& fromPos, const Position& toPos) const
+//get a line between two points
+//based on Xiaolin Wu algorithm
+std::vector< std::pair<int, int> > get2DLine(int x0, int y0, int x1, int y1)
 {
-	if (fromPos == toPos) {
-		return true;
+	std::vector< std::pair<int, int> > line;
+	if (x0 == x1 && y0 == y1) {
+		return line;
 	}
 
-	Position start(fromPos.z > toPos.z ? toPos : fromPos);
-	Position destination(fromPos.z > toPos.z ? fromPos : toPos);
+	bool isSteep = std::abs(y1 - y0) > std::abs(x1 - x0);
 
-	const int8_t mx = start.x < destination.x ? 1 : start.x == destination.x ? 0 : -1;
-	const int8_t my = start.y < destination.y ? 1 : start.y == destination.y ? 0 : -1;
+	if (isSteep) {
+		std::swap(x0, y0);
+		std::swap(x1, y1);
+	}
 
-	int32_t A = Position::getOffsetY(destination, start);
-	int32_t B = Position::getOffsetX(start, destination);
-	int32_t C = -(A * destination.x + B * destination.y);
+	if (x0 > x1) {
+		std::swap(x0, x1);
+		std::swap(y0, y1);
+	}
 
-	while (start.x != destination.x || start.y != destination.y) {
-		int32_t move_hor = std::abs(A * (start.x + mx) + B * (start.y) + C);
-		int32_t move_ver = std::abs(A * (start.x) + B * (start.y + my) + C);
-		int32_t move_cross = std::abs(A * (start.x + mx) + B * (start.y + my) + C);
+	float dx = x1 - x0;
+	float dy = y1 - y0;
+	float grad = 1.0;
 
-		if (start.y != destination.y && (start.x == destination.x || move_hor > move_ver || move_hor > move_cross)) {
-			start.y += my;
-		}
+	if (dx != 0) {
+		grad = dy * 1.0 / dx;
+		// We need to offset the line for isometric layout
+		grad = grad + .03;
+	}
 
-		if (start.x != destination.x && (start.y == destination.y || move_ver > move_hor || move_ver > move_cross)) {
-			start.x += mx;
-		}
+	float interY = y0 + grad; //first y - intersection for the main loop
 
-		const Tile* tile = getTile(start.x, start.y, start.z);
-		if (tile && tile->hasProperty(CONST_PROP_BLOCKPROJECTILE)) {
-			return false;
+	if (isSteep) {
+		for (int y = x0 + 1; y < x1; y++) {
+			int newX = std::floor(interY);
+			int newY = y;
+			line.push_back({ newX, newY });
+			interY += grad;
+		}
+	} else {
+		for (int x = x0 + 1; x < x1; x++) {
+			int newX = x;
+			int newY = std::floor(interY);
+			line.push_back({ newX, newY });
+			interY += grad;
```

Comment:
There will be speed benefits if you `reserve` space in `line` before `push_back`s. It will always be `x1 - x0` long. Also, use `emplace_back` instead.

By the way, `++x` is preferred over `x++`.